src/common/extrinsicService/useExtrinsic.tsx

```diff
@@ -25,100 +24,83 @@ export function useExtrinsic() {
   const { submitExtrinsic, estimateFee } = useExtrinsicProvider();
 
   async function sendTransaction({
-    destinationAddress,
     chainId,
-    transferAmmount,
     asset,
-    transferAll,
     keyring,
+    destinationAddress,
+    transferAmount,
+    transferAll,
   }: SendTransaction) {
     const address = decodeAddress(destinationAddress);
     const assetId = getAssetId(asset);
-    const args = {
-      dest: address,
-      value: transferAmmount,
-      asset: assetId,
-    };
     let signOptions;
     let transactionType = TransactionType.TRANSFER;
 
     if (transferAll) {
       transactionType = TransactionType.TRANSFER_ALL;
     }
-    if (asset?.type === AssetType.STATEMINE) {
+    if (asset.type === AssetType.STATEMINE) {
       transactionType = TransactionType.TRANSFER_STATEMINE;
       signOptions = getAssetIdSignOption(assetId);
     }
 
-    return await submitExtrinsic({
-      chainId: chainId,
+    return submitExtrinsic({
+      chainId,
+      signOptions,
+      keyring,
       transaction: {
         type: transactionType,
-        args,
+        args: {
+          dest: address,
+          value: transferAmount,
+          asset: assetId,
+        },
       },
-      signOptions,
-      keyring,
     }).then(hash => {
-      if (!hash) {
-        throw Error('Something went wrong');
-      }
+      if (!hash) throw Error('Something went wrong');
+
       console.log('Success, Hash:', hash?.toString());
     });
   }
 
-  async function handleSendGift(
-    { chainId, amount, transferAll, asset, fee }: TrasferAsset,
+  async function sendGift(
+    { chainId, amount, transferAll, asset }: TrasferAsset,
     giftTransferAddress: string,
-  ) {
-    const transferAmmount = formatAmount(amount as string, asset.precision);
-
-    if (isStatemineAsset(asset?.type)) {
-      // Fee is 2x the transfer fee - we add 1 fee to the transfer amount
-      const giftAmount = Math.ceil(+transferAmmount + fee! / 2).toString();
+  ): Promise<void> {
+    const transferAmount = Number(formatAmount(amount!, asset.precision));
+    const trasferAllFee = await getTransactionFee(chainId, TransactionType.TRANSFER_ALL);
+    const giftAmount = (transferAmount + trasferAllFee).toString();
 
-      return await sendTransaction({
-        destinationAddress: giftTransferAddress,
-        chainId,
-        transferAmmount: giftAmount,
-        asset,
-      });
-    }
```

Comment:
We still need this part to calculate fees correctly since we don't have transferAll for statemine